lang/printer/src/render/html.rs

```diff
@@ -0,0 +1,63 @@
+use std::io;
+
+use crate::types::*;
+
+pub struct RenderHtml<W> {
+    anno_stack: Vec<Anno>,
+    upstream: W,
+}
+
+impl<W> RenderHtml<W> {
+    pub fn new(upstream: W) -> RenderHtml<W> {
+        RenderHtml { anno_stack: Vec::new(), upstream }
+    }
+}
+
+impl<W> pretty::Render for RenderHtml<W>
+where
+    W: io::Write,
+{
+    type Error = io::Error;
+
+    fn write_str(&mut self, s: &str) -> io::Result<usize> {
+        self.upstream.write(s.as_bytes())
+    }
+
+    fn write_str_all(&mut self, s: &str) -> io::Result<()> {
+        self.upstream.write_all(s.as_bytes())
+    }
+
+    fn fail_doc(&self) -> Self::Error {
+        io::Error::new(io::ErrorKind::Other, "Document failed to render")
+    }
+}
+
+impl<W> pretty::RenderAnnotated<'_, Anno> for RenderHtml<W>
+where
+    W: io::Write,
+{
+    fn push_annotation(&mut self, anno: &Anno) -> Result<(), Self::Error> {
+        self.anno_stack.push(*anno);
+        let out = match anno {
+            Anno::Keyword => "<span class=\"keyword\">",
+            Anno::Ctor => "<span class=\"ctor\">",
+            Anno::Dtor => "<span class=\"dtor\">",
+            Anno::Type => "<span class=\"type\">",
+            Anno::Comment => "<span class=\"comment\">",
+            Anno::Backslash => "<span class=\"backslash\">",
```

Comment:
I don't think we need those cases for `Anno::Backslash`, `Anno::BraceOpen` and `Anno::BraceClose`. These are mainly necessary for the latex backend, because when we generate latex we have to escape them to generate valid tex code.